blog/2019-02-07-KIEServerOptaPlannerTaskAssignment.adoc

```diff
@@ -0,0 +1,215 @@
+    {
+      'create-container': {
+        'container': {
+          'container-id': CONTAINER_ID,
+          'release-id': {
+            'group-id': GROUP_ID,
+            'artifact-id': ARTIFACT_ID,
+            'version': VERSION,
+          },
+        },
+      },
+    },
+  ],
+}
+----
+Notice that the GAV in the `release-id` object are your KJAR's GAV.
+
+=== Solver registration
+
+`PUT /containers/{CONTAINER_ID}/solvers/{SOLVER_ID}`
+
+With body:
+[source,json,options="nowrap"]
+----
+{
+  'solver-config-file': 'PATH_TO_SOLVER_CONFIG_FILE.xml'
+}
+----
+This will build a new solver from the xml resource included in the KJAR.
+
+=== Submit a problem
+
+`POST /containers/{CONTAINER_ID}/solvers/{SOLVER_ID}/state/solving`
+
+Once a solver is built it will be waiting for a problem to start solving. The body of this request contains the object
+annotated as `@PlanningSolution`, in the Task Assignment example it will be `TaskAssigningSolution`.
+
+=== Query bestSolution
+
+`GET /containers/{CONTAINER_ID}/solvers/{SOLVER_ID}/bestsolution`
+
+This response body will contain `best-solution` object in addition to extra information about the solver status and score.
+
+=== Submit a ProblemFactChange
+
+`POST /containers/{CONTAINER_ID}/solvers/{SOLVER_ID}/problemfactchanges`
+
+Submits a ProblemFactChange to update the problem the solver is solving. For example if you want to delete
+a task the body should be:
+//TODO: Update the example to JSON once figure out the required format
+[source,xml,options="nowrap"]
+----
+<problem-fact-change class="TaDeleteTaskProblemFactChange">
+	<taskId>TO_BE_DELETED_TASK_ID</taskId>
+</problem-fact-change>
+----
+Notic the use of `class` attribute, this is how you tell OptaPlanner service what type of ProblemFactChange you are submitting.
+Here `TaDeleteTaskProblemFactChange` is an `@XStreamAlias` for the `DeleteTaskProblemFactChange`.
+
+
+NOTE: All the requests above have a base URL `http://SERVER:PORT/CONTEXT/services/rest/server` and require basic HTTP
+Authentication for the role kie-server.
+
+You can check all the available endpoints https://docs.optaplanner.org/7.14.0.Final/optaplanner-wb-es-docs/html_single/#_optaplanner_rest_api[here]
+
+
+== Conclusion
+
+To integrate an OptaPlanner service with your application on KIE Server:
+
+1. Build the service as a KJAR.
+2. Send an HTTP request to the KIE Server to start a container that runs this service.
+3. Communicate with the service through the REST API the the KIE Server exposes.
+
+
+== Related material
+
+https://github.com/kiegroup/kie-server-optaplanner-task-assignment[KIE Server OptaPlanner Task Assignment demo]
+
+https://docs.optaplanner.org/7.17.0.Final/optaplanner-wb-es-docs/html_single/#kie-server-commands-con_kie-apis[KIE Server and KIE container commands in OptaPlanner]
+
+//TODO: update the link to 7.17.0.Final
+https://docs.optaplanner.org/7.14.0.Final/optaplanner-wb-es-docs/html_single/#_optaplanner_rest_api[OptaPlanner REST API]
```

Comment:
Maybe let's use https://docs.optaplanner.org/latestFinal/ everywhere? Wdyt, @ge0ffrey?

Reply:
+100 :)

Reply:
The problem this section disappeared since 7.14.0.Final, I submitted an issue for it, but I agree with using latestFinal for all other links